x-pack/plugins/cloud_security_posture/server/lib/fleet_util.ts

```diff
@@ -43,17 +44,27 @@ export type AgentStatusByAgentPolicyMap = Record<string, GetAgentStatusResponse[
 
 export const getAgentStatusesByAgentPolicies = async (
   agentService: AgentService,
-  agentPolicies: AgentPolicy[] | undefined
+  agentPolicies: AgentPolicy[] | undefined,
+  logger: Logger
 ): Promise<AgentStatusByAgentPolicyMap> => {
   if (!agentPolicies?.length) return {};
 
   const internalAgentService = agentService.asInternalUser;
   const result: AgentStatusByAgentPolicyMap = {};
 
-  for (const agentPolicy of agentPolicies) {
-    result[agentPolicy.id] = await internalAgentService.getAgentStatusForAgentPolicy(
-      agentPolicy.id
-    );
+  try {
+    for (const agentPolicy of agentPolicies) {
+      result[agentPolicy.id] = await internalAgentService.getAgentStatusForAgentPolicy(
+        agentPolicy.id
+      );
+    }
+  } catch (error) {
+    const httpError = t.type({ statusCode: t.number });
```

Comment:
1. consider renaming to `fleetHttpError`, as it is more descriptive and as we're catching a fleet service error that adds http data, it seems more reasonable. 
2. please define the schema on a module scope so it won't be re-created every time the catch block is hit.

Reply:
> please define the schema on a module scope so it won't be re-created every time the catch block is hit.

🚀